File: 69.sqrt-x.js
/*
 * @lc app=leetcode id=69 lang=javascript
 *
 * [69] Sqrt(x)
 */

// @lc code=start
/**
 * @param {number} x
 * @return {number}
 */
var mySqrt = function(x) {
  r = x;
  while (r * r > x) {
    r = ((r + x / r) / 2) | 0;
  }
  return r;
};
// @lc code=end
// 1017/1017 cases passed (68 ms)
// Your runtime beats 87.8 % of javascript submissions
// Your memory usage beats 72.22 % of javascript submissions (35.6 MB)

File: 69.sqrt-x.java
/*
 * @lc app=leetcode id=69 lang=java
 *
 * [69] Sqrt(x)
 */

// @lc code=start
class Solution {
    public int mySqrt(int x) {
        long r = x;
        while (r * r > x) {
            r = (r + x / r) / 2;
        }
        return (int) r;
    }
}
// @lc code=end
// 1017/1017 cases passed (1 ms)
// Your runtime beats 100 % of java submissions
// Your memory usage beats 5 % of java submissions (33.7 MB)